lib/backup/azremote/azblob_test.go

```diff
@@ -0,0 +1,190 @@
+package azremote
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+)
+
+func Test_FSInit(t *testing.T) {
+	cases := map[string]struct {
+		IgnoreFakeEnv bool
+		Env           testEnv
+		Dir           string
+		ExpectedDir   string
+		ExpectedErr   string
+		ExpectedLogs  []string
+	}{
+		"dir / prefix is removed": {
+			Dir:         "/foo/",
+			ExpectedDir: "foo/",
+			ExpectedErr: "failed to detect any credentials",
+		},
+		"multiple dir prefix / is removed": {
+			Dir:         "//foo/",
+			ExpectedDir: "foo/",
+			ExpectedErr: "failed to detect any credentials",
+		},
+		"suffix is added": {
+			Dir:         "foo",
+			ExpectedDir: "foo/",
+			ExpectedErr: "failed to detect any credentials",
+		},
```

Comment:
Could you split tests for `Dir` verification and `Client` init into separate functions?
That allows to narrow the test cases scope and makes code easier to read.

Reply:
93ed9423d